teamcity: failed test: TestLogic

The following tests appear to have failed on master (test): TestLogic/fakedist: TestLogic/fakedist-metadata: TestLogic/fakedist-metadata/rename_table, TestLogic, TestLogic/fakedist: TestLogic/fakedist-metadata

You may want to check [for open issues](https://github.com/cockroachdb/cockroach/issues?q=is%3Aissue+is%3Aopen+TestLogic).

[#1165064](https://teamcity.cockroachdb.com/viewLog.html?buildId=1165064):


```
TestLogic/fakedist: TestLogic/fakedist-metadata
--- FAIL: test/TestLogic: TestLogic/fakedist: TestLogic/fakedist-metadata (0.350s)




TestLogic/fakedist: TestLogic/fakedist-metadata: TestLogic/fakedist-metadata/rename_table
--- FAIL: test/TestLogic: TestLogic/fakedist: TestLogic/fakedist-metadata: TestLogic/fakedist-metadata/rename_table (2.010s)

------- Stdout: -------
=== PAUSE TestLogic/fakedist-metadata/rename_table
            logic.go:2329: 
                
                testdata/logic_test/rename_table:230: 
                expected success, but found
                pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE): "sql txn" id=8323ec0a key=/Table/SystemConfigSpan/Start rw=true pri=0.01679640 stat=PENDING epo=0 ts=1551915861.033914937,1 orig=1551915861.033674837,0 max=1551915861.033674837,0 wto=false seq=19
            logic.go:2345: 
                testdata/logic_test/rename_table:230: error while processing
            logic.go:2346: pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE): "sql txn" id=8323ec0a key=/Table/SystemConfigSpan/Start rw=true pri=0.01679640 stat=PENDING epo=0 ts=1551915861.033914937,1 orig=1551915861.033674837,0 max=1551915861.033674837,0 wto=false seq=19



TestLogic
--- FAIL: test/TestLogic (444.500s)

------- Stdout: -------
    test_log_scope.go:81: test logs captured to: /tmp/logTestLogic176002569
    test_log_scope.go:62: use -show-logs to present logs inline




```

Please assign, take a look and update the issue accordingly.

--- a/pkg/sql/logictest/logic.go
+++ b/pkg/sql/logictest/logic.go
@@ -1025,7 +1025,9 @@ func (t *logicTest) setup(cfg testClusterConfig) {
 	}
 
 	// Update the defaults for automatic statistics to avoid delays in testing.
-	stats.DefaultAsOfTime = time.Microsecond
+	// Avoid making the DefaultAsOfTime too small to avoid interacting with
+	// schema changes and causing transaction retries.
+	stats.DefaultAsOfTime = 10 * time.Millisecond
 	stats.DefaultRefreshInterval = time.Millisecond
 
 	t.cluster = serverutils.StartTestCluster(t.t, cfg.numNodes, params)

